### PR TITLE
bug/change caching scope on auth calls

### PR DESCRIPTION
**Pull request by jimPruyne**
- Change the behavior of caching for Auth/Groups calls
- Add optimization for principal check without groups

### CODE REVIEW

File: globus_action_provider_tools/authentication.py [hunk 1]
```diff
@@ -106,10 +119,13 @@ def principals(self) -> FrozenSet[str]:
...
 
     @property  # type: ignore
-    @dogpile_cache.cache_on_arguments()
     def groups(self) -> FrozenSet[str]:
         try:
             groups_client = self._get_groups_client()
+            groups_token = groups_client.authorizer.access_token
+            groups_set = AuthState.group_membership_cache.get(groups_token)
+            if groups_set is not None:
+                return groups_set
```
[the remark this answers is not in the document]
Comment (jimPruyne) [reply globus_action_provider_tools/authentication.py:128]: ok

File: globus_action_provider_tools/authentication.py [hunk 2]
```diff
@@ -166,14 +198,29 @@ def _get_groups_client(self) -> GroupsClient:
...
         allow_all_authenticated_users: bool = False,
     ) -> bool:
         allowed_set = frozenset(allowed_principals)
-        all_principals = self.identities.union(self.groups)
+        all_principals = self.identities
+        # We only need to merge in the groups values to the principals list if there are
+        # group principals in the list. Can save a round trip to the Groups service if
+        # there's no need to check for group membership.
+        if AuthState.group_in_principal_list(allowed_set):
+            allowed_principals = set(allowed_principals).union(self.groups)
```
[the remark this answers is not in the document]
Comment (jimPruyne) [reply globus_action_provider_tools/authentication.py:223]: came up in one of our discussions. Not sure who originally pointed it out.

**Comment by jimPruyne** [comment]
@urimandujano  Thanks for suggesting adding the test. I've got something there for it now, though it might not be the most robust test?

**Review by urimandujano — APPROVED** [review]
Looks good

**Pull request merged**